homeassistant/components/madvr/config_flow.py

```diff
@@ -0,0 +1,90 @@
+"""Config flow for the integration."""
+
+import asyncio
+import logging
+from typing import Any
+
+import aiohttp
+from madvr.errors import CannotConnect
+from madvr.madvr import HeartBeatError, Madvr
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_PORT
+
+from .const import DEFAULT_NAME, DEFAULT_PORT, DOMAIN
+from .coordinator import MadVRCoordinator
+
+type MadVRConfigEntry = ConfigEntry[MadVRCoordinator]
+
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(
+            CONF_HOST,
+        ): str,
+        vol.Required(
+            CONF_PORT,
+            default=DEFAULT_PORT,
+        ): int,
+    }
+)
+
+
+class MadVRConfigFlow(ConfigFlow, domain=DOMAIN):
```

Comment:
Again the question, is there something unique we can get from the device? serial number, macaddress etc?

Reply:
once the connection is established it is possible to ask it for a mac address but nothing before a connection is established. What is the purpose of unique info here?

Reply:
Making sure someone can't setup the same config flow twice

Reply:
before it was asking for a name and mac, but I was told to remove that. So I really don't know what else to do here.

Reply:
Oh I misread that, I read that we could retrieve the Mac.

Can we acquire something from the device that is unique after connecting?

Reply:
nothing besides a mac there is no command to get any unique info

Could ask the user to input their serial number but at the same time, it doesn't really matter if you set it up twice because the connections will be completely independent, I tested this with multiple instances of the integration

Reply:
Yea but it's a user experience thing. In that case we should abort if we see the same host.

Reply:
okay I added 

```
await self.async_set_unique_id(host)
self._abort_if_unique_id_configured()
```